packages/jupyter-chat/src/components/chat-input.tsx

```diff
@@ -17,33 +18,27 @@ import React, { useEffect, useRef, useState } from 'react';
 
 import { AttachmentPreviewList } from './attachments';
 import { AttachButton, CancelButton, SendButton } from './input';
-import { IChatModel } from '../model';
+import { IInputModel, InputModel } from '../input-model';
 import { IAutocompletionRegistry } from '../registry';
-import { IAttachment, IConfig, Selection } from '../types';
+import { IAttachment, Selection } from '../types';
 import { useChatCommands } from './input/use-chat-commands';
 import { IChatCommandRegistry } from '../chat-commands';
 
 const INPUT_BOX_CLASS = 'jp-chat-input-container';
 
 export function ChatInput(props: ChatInput.IProps): JSX.Element {
   const { documentManager, model } = props;
-  const [input, setInput] = useState<string>(props.value || '');
+  const [input, setInput] = useState<string>(model.value);
```

Comment:
I'm hesitant to leave this line here because it creates 2 sources of truth about what the current state of the input is (`model.value` v.s. `input`). It seems like we should remove the `input` and `setInput` variables entirely. Can you explore making these changes?

Reply:
As discussed privately, the state is used to update the react component.
`setInput()` should only be called when the input model value has changed, I don't think that it can lead to errors.

Reply:
I agree. I did 30 min of research into React rendering and it doesn't seem like there's any other way to trigger a re-render. This leads to a little duplicate state, but I think that's fine given that there's no clear alternative.